frontend/apps/reader/readerui.lua

```diff
@@ -464,23 +464,26 @@ function ReaderUI:init()
     end
     self.postInitCallback = nil
 
-    -- Now that document is loaded, store book metadata in settings
-    -- (so that filemanager can use it from sideCar file to display
-    -- Book information).
-    local props = self.document:getProps()
-    self.doc_settings:saveSetting("doc_props", props)
+    -- Now that document is loaded, store book metadata in settings.
+    local props = self.doc_settings:readSetting("doc_props")
+    if props == nil then
+        props = self.document:getProps()
```

Comment:
The previous code was always updating doc_props with the fresh metadata from the opened document, which were available for free - so it always felt to me it was a good idea in case one has modified/fixed the document (which I sometimes do on the emulator). May be this may not really happen in practice, but dunno, may be with calibre updating books and metadata?
Any real reason to prefer keeping the first seen metadata, and possibly having stalled metadata? (I guess there could be reaons, like losing statistics if title/authors is updated, but I dunno which is best - so want to know why for you prefer it this way.)

Reply:
To speed up opening a document.
If you feel that the original metadata is sometimes edited, I will revert.

Reply:
> To speed up opening a document.

Feels like at this point (book opened), the metadata is already parsed and just ready to use.

> If you feel that the original metadata is sometimes edited, I will revert.

I'd like more feedback so we can make the best decision.
Either we keep it as first seen and no issue I guess (but possibly staled info).
Either we update it (as we've done for years), and we may get some issues like statistics no long found (but we haven't had any report for years).

> If metadata is edited, md5 should be recalculated every time as well.

I'm less worried about this one: it's just some random ID we compute once and store and reuse what was stored - so best if it's stable no matter how metadata have changed.
(Except may be for the new hashdocsettings, and if docsettings is Reset in File browser while one would like to keep statistics.)

Reply:
@Frenzie @NiLuJe : thoughts on this and these _Either/Either_ ?

Reply:
I agree with @poire-z here that you might as well always update it.

But either/either what?

Reply:
The BookInfo cache does *not* pick up metadata changes automatically, so, I feel like not picking 'em here in DocSettings would track with that. There's always the "Refresh metadata" button when you *know* you've updated metadata for that book ;).

Reply:
Then again, as @poire-z mentioned, it's essentially free to pick 'em up, so, eeeeeh ;).